webapi/src/main/scala/org/knora/webapi/routing/v1/ValuesRouteV1.scala

```diff
@@ -195,12 +200,44 @@ object ValuesRouteV1 extends Authenticator {
         ValueGetRequestV1(valueIri, userProfile)
     }
 
+    private def makeChangeFileValueRequest(resourceIri: IRI, apiRequest: Option[ChangeFileValueApiRequestV1], multipartConversionRequest: Option[SipiResponderConversionPathRequestV1], userProfile: UserProfileV1) = {
+        if (apiRequest.nonEmpty && multipartConversionRequest.nonEmpty) throw BadRequestException("File information is present twice, only one is allowed.")
+
+        if (apiRequest.nonEmpty) {
+            // GUI-case
+            val fileRequest = SipiResponderConversionFileRequestV1(
+                originalFilename = InputValidation.toSparqlEncodedString(apiRequest.get.file.originalFilename),
+                originalMimeType = InputValidation.toSparqlEncodedString(apiRequest.get.file.originalMimeType),
+                filename = InputValidation.toSparqlEncodedString(apiRequest.get.file.filename),
+                userProfile = userProfile
+            )
+            ChangeFileValueRequestV1(
+                resourceIri = resourceIri,
+                file = fileRequest,
+                apiRequestID = UUID.randomUUID,
+                userProfile = userProfile)
+        }
+        else if (multipartConversionRequest.nonEmpty) {
+            // non GUI-case
+            ChangeFileValueRequestV1(
+                resourceIri = resourceIri,
+                file = multipartConversionRequest.get,
+                apiRequestID = UUID.randomUUID,
+                userProfile = userProfile)
+        } else {
+            // no file information was provided
+            throw BadRequestException("A file value change was requested but no file information was provided")
+        }
+
+    }
+
     def rapierPath(_system: ActorSystem, settings: SettingsImpl, log: LoggingAdapter): Route = {
         implicit val system = _system
         implicit val executionContext = system.dispatcher
         implicit val timeout = settings.defaultTimeout
         val responderManager = system.actorSelection("/user/responderManager")
 
+        // TODO: add documentation
```

Comment:
Haha, actually this todo was meant for you because I did not get how this is supposed to work without additional information (the client has to submit three Iris) :-)